src/index.js

```diff
@@ -183,7 +183,12 @@ const userEvent = {
     };
     const opts = Object.assign(defaultOpts, userOpts);
 
-    const computedText = text.slice(0, element.maxLength || text.length);
+    let computedText;
+    if (!element.maxLength || element.maxLength === -1) {
+      computedText = text.slice(0, text.length);
```

Comment:
This could be simplified to

```suggestion
      computedText = text.slice(0, element.maxLength > 0 ? element.maxLength : text.length);
```

and avoid the conditional assignment. It's also slightly more spec conformant since any invalid value should result in no max length being applied (realizing that jsdom diverges by using `0` as its default/unlimited value)